docs/source/contributor-guide/index.md

```diff
@@ -66,24 +66,33 @@ ideas with the community to get feedback on implementation.
 
 ## Pull Request Overview
 
-We welcome pull requests (PRs) from anyone from the community.
+We welcome pull requests (PRs) from anyone in the community.
 
-DataFusion is a very active fast-moving project and we try to review and merge PRs quickly to keep the review backlog down and the pace up. After review and approval, one of the [committers] will merge your PR.
+DataFusion is a very active fast-moving project and we try to review and merge PRs quickly.
 
 Review bandwidth is currently our most limited resource, and we highly encourage reviews by the broader community. If you are waiting for your PR to be reviewed, consider helping review other PRs that are waiting. Such review both helps the reviewer to learn the codebase and become more expert, as well as helps identify issues in the PR (such as lack of test coverage), that can be addressed and make future reviews faster and more efficient.
 
+The lifecycle of a PR is:
+
+1. Create a PR targeting the `main` branch.
+2. For returning contributors, CI tests are automatically run. For new contributors a committer must first trigger the tests.
+3. Your PR will be reviewed. Please respond to all feedback on the PR: you don't have to change the code, but you should acknowledge the feedback.
+4. Once the PR is approved, one of the [committers] will merge your PR, typically within 24 hours. We leave approved "major" changes (see below) open for 24 hours prior to merging, and sometimes leave "minor" PRs open for the same time to permit additional feedback.
```

Comment:
```suggestion
4. Once the PR is approved, one of the [committers] will merge your PR, typically within 24 hours. We leave approved "major" changes (see below) open for 24 hours prior to merging, and sometimes leave "minor" PRs open for the same time to permit additional feedback. Besides, above time frame is an estimate typically. Due to limited bandwidth of committers, it might take longer to get your approved PR merged. Please wait patiently. If it takes too longer, e.g., many days, you can friendly ping the committer who approved your PR to get their notice once they forget to merge it.
```

Reply:
in https://github.com/apache/datafusion/pull/10532/commits/e70f0291ac4b14592a6bec51dcd365f30d1246f2
Thank you